src/Controller/EloquaAppCloudEndpointController.php

```diff
@@ -0,0 +1,303 @@
+    $response->requiresConfiguration = FALSE;
+    \Drupal::logger('EloquaAppCloudDecision')->notice(print_r($response, TRUE));
+    return new JsonResponse($response);
+  }
+
+  /**
+   * Delete.
+   *
+   * @return string
+   *   Return Hello string.
+   */
+  public function delete($eloqua_app_cloud_service) {
+  // Delete any existing queue entries for this service entity.
+
+  }
+
+  /**
+   * Execute.
+   *
+   * @return string
+   *   Return Hello string.
+   */
+  public function execute($eloqua_app_cloud_service) {
+    \Drupal::logger('EloquaAppCloudDecision')->notice('Executing' );
+    // Get the instanceID from the query parameter.
+    $instanceId = $this->request->get("instance");
+    // Get the execution ID
+    $executionId = $this->request->get("executionId");
+    if (empty($instanceId)) {
+      // @TODO Throw an exception, and/or return an error?
+      return new JsonResponse(['error no instanceId']);
+    }
+    //
+    //$content = $this->mockContent();
+    // Now load the JSON payload form Eloqua.
+     $content = $this->request->getContent();
+
+    if (empty($content)) {
+      // @TODO Throw an exception, and/or return an error?
+      return new JsonResponse(['error no content']);
+    }
+    $payload = json_decode($content);
+    $records = $payload->items;
+
+    $pluginReferences = $this->getEntityPlugins($eloqua_app_cloud_service);
+
+    // Iterate over the ServiceEntity plugins and then over the payload items.
+    foreach ($pluginReferences as $pluginReference) {
+      $id = $pluginReference->value;
+      // Get the plugin manager from the list of plugins (from the container).
+      $pluginMgr = $this->plugins[$id];
+      // Instantiate the referenced plugin.
+      $plugin = $pluginMgr->createInstance($id);
+
+      /**
+       * Get the appropriate queue for this plugin.
+       * @var QueueInterface $queue
+       */
+      $queue = $this->queueFactory->get($plugin->queueWorker());
+        // Put the records directly onto on the queue.
+      foreach ($records as $record) {
+        // Put the result of the plugin on each record.
+        $record->result = $plugin->execute($record);
+      }
+      // @TODO Define a queueItem class?
+      $queueItem = new \stdClass();
+      // Pass the queue type to the worker to make it easy to requeue if there are more then 5000 records.
+      $queueItem->queueId = $plugin->queueWorker();
+      // Pass the instance ID so the worker can communicate with Eloqua.
+      $queueItem->instanceId = $instanceId;
+      if(!empty($executionId)){
+        $queueItem->executionId = $executionId;
+      }
+      $queueItem->api = $plugin->api();
+      $queueItem->fieldList = $plugin->fieldList();
+      $queueItem->records = $records;
+      $queue->createItem($queueItem);
+    }
+
+    // If we have gotten through all that we just return a 204 to indicate an asynchronous response.
+    $response = [];
+    $response['statue'] = 204;
```

Comment:
Statue

Reply:
Also, thought this would need to be HTTP status code (`return new JsonResponse(NULL, 204)`, or is this some kind of magic Symfony thing?